server/src/main/java/org/opensearch/indices/replication/SegmentReplicationSource.java

```diff
@@ -49,6 +49,13 @@ void getSegmentFiles(
         ActionListener<GetSegmentFilesResponse> listener
     );
 
+    /**
+     * Get the source description
+     */
+    default String getSourceDescription() {
```

Comment:
Given this is going to be displayed directly in a rest API, I do not think we should provide a default here and require an implementation.